File: springboot-backend/src/main/java/backend/dto/response/OrderResponse.java
package backend.dto.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class OrderResponse {
    private Long id;
    private Double totalPrice;
    private String status;
    private String shippingAddress;
    private String paymentMethod;
    private String createdAt;
    
    private List<OrderItemResponse> items;

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class OrderItemResponse {
        private Long productId;
        private String productName;
        private Integer quantity;
        private double price;
        private String productImage;
    }
}

File: Fontend/app/src/main/java/com/example/data/dto/response/OrderResponse.kt
package com.example.data.dto.response

import com.google.gson.annotations.SerializedName

data class OrderResponse(
    @SerializedName("id") val id: Long,
    @SerializedName("totalPrice") val totalPrice: Double,
    @SerializedName("status") val status: String?,
    @SerializedName("shippingAddress") val shippingAddress: String?,
    @SerializedName("paymentMethod") val paymentMethod: String?,
    @SerializedName("createdAt") val createdAt: String?,
    @SerializedName("items") val items: List<OrderItemResponse>?
)
data class OrderItemResponse(
    @SerializedName("productId") val productId: Long,
    @SerializedName("productName") val productName: String?,
    @SerializedName("quantity") val quantity: Int,
    @SerializedName("price") val price: Double,
    @SerializedName("productImage") val productImage: String?
)